lib/src/api/engine/mod.rs

```diff
@@ -13,6 +13,7 @@ pub mod any;
 pub mod local;
 #[cfg(any(feature = "protocol-http", feature = "protocol-ws"))]
 pub mod remote;
+pub mod tasks;
```

Comment:
This should be hidden. We shouldn't expose it as part of public API.

Reply:
Tbh I think I am going to move it to the core. It is a useful struct to have for debugging. Happy to put a cfg(test_assertions) either on declaration or operations. I may be useful to have it in the release builds though as well. Case being the future tasks that are spawned

If this were to go to core, where do you think is should go? And it would be pub exposed then, but will put a doc hidden on it

Reply:
That needs to be exposed because its used by the CLI. Adding a docs hidden

Reply:
Yes, I meant exposed publicly in the docs.